tensorflow/core/user_ops/gather_columns_op_test.py

```diff
@@ -0,0 +1,195 @@
+                                                 np.array(params) * row3]), dtype=dtype)
+                else:
+                    params_matrix = np.empty([self.num_rows, self.num_cols])
+                    params_row = np.array(params)
+                    for i in range(0, self.num_rows):
+                        params_matrix[i,:] = params_row * (i+1)
+                    p2d2 = tf.constant(params_matrix, dtype=dtype)
+
+                    # For testing only the overhead time
+                    #p2d2 = tf.constant(params, dtype=dtype)
+
+                op1d = self.gather_columns_module.gather_columns(p1d, indices)
+                op2d1 = self.gather_columns_module.gather_columns(p2d1, indices)
+                op2d2 = self.gather_columns_module.gather_columns(p2d2, indices)
+
+                with tf.Session() as sess:
+                    out1d = sess.run(op1d)
+                    out2d1 = sess.run(op2d1)
+                    out2d2 = sess.run(op2d2)
+
+                np.testing.assert_array_almost_equal(out1d, true_output)
+                self.assertEqual(dtype.as_numpy_dtype, out1d.dtype)
+
+                true_output_2d1 = [np.array(true_output)]
+                np.testing.assert_array_almost_equal(out2d1, true_output_2d1)
+                self.assertEqual(dtype.as_numpy_dtype, out2d1.dtype)
+
+                if not large_case:
+                    true_output_2d2 = [np.array(true_output) * row1,
+                                       np.array(true_output) * row2,
+                                       np.array(true_output) * row3]
+                else:
+                    true_output_row = np.array(true_output)
+                    for i in range(0, self.num_rows):
+                        params_matrix[i,:] = true_output_row * (i+1)
+                    true_output_2d2 = params_matrix
+
+                    # For testing only the overhead time
+                    #true_output_2d2 = true_output
+
+                np.testing.assert_array_almost_equal(out2d2, true_output_2d2)
+                self.assertEqual(dtype.as_numpy_dtype, out2d2.dtype)
+
+
+        float_val = 1.23456789
+        int_val = 123456789
+        int_32_upper = 2147483647
+        int_64_upper = 9223372036854775807
+
+
+        # Single column input tensor
+        # float
+        test([float_val],
+             [0],
+             tf.float32,
+             [float_val])
+        test([float_val],
+             [0],
+             tf.float64,
+             [float_val])
+
+        # int
+        test([int_32_upper],
+             [0],
+             tf.int32,
+             [int_32_upper])
+        test([int_64_upper],
+             [0],
+             tf.int64,
+             [int_64_upper])
+
+        # bool
+        test([True],
+             [0],
+             tf.bool,
+             [True])
+
+        # Single index
+        # float
+        test([float_val, float_val*2, float_val*3],
```

Comment:
Please generally add a test for testing exceptions if indices are outside the range